fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java

```diff
@@ -170,13 +169,58 @@ public boolean visit(java.lang.String elementName, int elementIndex, com.ibm.fhi
             if (!REFERENCE.equals(searchParamType) && !URI.equals(searchParamType)) {
                 throw invalidComboException(searchParamType, canonical);
             }
-            StringParmVal p = new StringParmVal();
-            p.setResourceType(resourceType);
-            p.setName(searchParamCode);
-            p.setUrl(searchParamUrl);
-            p.setVersion(searchParamVersion);
-            p.setValueString(canonical.getValue());
-            result.add(p);
+            
+            // For REFERENCE search parameters, we need to treat as both string and as composite.
+            if (REFERENCE.equals(this.searchParamType)) {
+                // Parse the canonical value into a uri and a version. Create a normal string
+                // parameter for the input canonical value, and then create as composite parm to correlate
+                // uri and version (even if version is null).
+                CanonicalValue canonicalValue = CanonicalSupport.createCanonicalValueFrom(canonical.getValue());
+                
+                // Just the uri parm
+                StringParmVal p = new StringParmVal();
```

Comment:
Since this StringParmValue extraction is duplicated in the "else" block, perhaps just move this part outside this entire "if" block.

Reply:
Done.